views/attendance/new_class.php

```diff
@@ -0,0 +1,291 @@
+    var classesMatrix = [
+        <?php
+            while($row = pg_fetch_assoc($result_classes)){
+                echo "[{$row['curriculumid']},\"". addslashes($row['topicname']) . "\", {$row['classid']}],";
+            }
+        ?>
+        ];
+
+    //js for controlling the disabled selection of class section
+    function enableSecondSelection() {
+
+        //disable submit button in case first class was changed
+        document.getElementById("sub").disabled = true;
+
+        //enable selection
+        document.getElementById("classSelection").disabled = false;
+
+        /* Display the proper classes */
+
+        //clear the current class selection
+        var classesElement = document.getElementById("classes");
+
+        while(classesElement.firstChild){
+            classesElement.removeChild(classesElement.firstChild);
+        }
+
+        //get current input of curriculum
+        var curriculumNumberSelected = $("#curr").find("option:selected").data("id");
+        document.getElementById('curr-id').value = curriculumNumberSelected;
+
+        //add new options
+
+        var node = document.createElement("OPTION");
+        node.selected = true;
+        node.disabled = true;
+        node.innerHTML = "Select Class";
+        classesElement.appendChild(node);
+
+        for(var i = 0; i < classesMatrix.length; i++){
+            if(classesMatrix[i][0] === curriculumNumberSelected){ //same course number
+                var classNode = document.createElement("OPTION");
+                classNode.setAttribute('id', 'top-' + classesMatrix[i][2].toString());
+                classNode.innerHTML = classesMatrix[i][1];
+                classNode.dataset.id = classesMatrix[i][2];
+                classesElement.appendChild(classNode);
+            }
+        }
+
+    }
+
+    function populateTimes() {
+        var selectTime = document.getElementById('time-input');
+
+        var optionArray = [];
+
+        //AM
+        for(var i = 0; i < 12; i++){
+            var hourAM;
+            if(i === 0){
+                hourAM = 12;
+            } else{
+                hourAM = i;
+            }
+            optionArray.push(createTime(hourAM, "00", "AM"))
+            optionArray.push(createTime(hourAM, "30", "AM"))
+        }
+
+        //PM
+        for(var j = 0; j < 12; j++){
+            var hourPM;
+            if(j === 0){
+                hourPM = 12;
+            } else{
+                hourPM = j;
+            }
+            optionArray.push(createTime(hourPM, "00", "PM"))
+            optionArray.push(createTime(hourPM, "30", "PM"))
+        }
+
+        for(var k = 0; k < optionArray.length; k++){
+            selectTime.appendChild(optionArray[k]);
+        }
+    }
+
+    //input: hour and time integers
+    //output: option object
+    function createTime(hour, minute, amORpm){
+        var option = document.createElement('OPTION');
+        option.innerHTML = (hour + ":" + minute + " " + amORpm).toString();
```

Comment:
`toString()` shouldn't be necessary here. `typeof(3+"AM")` is `"string"`

Reply:
for readability cause there's a lot of messed up typing in JS